app/src/main/java/com/github/se/polyfit/ui/components/ingredients/IngredientNutritionEditFields.kt

```diff
@@ -109,3 +105,34 @@ fun IngredientNutritionEditFields(
     }
   }
 }
+
+private fun tryCatchBlock(
+    index: Int,
+    nutritionFields: MutableList<Nutrient>,
+    nutrient: Nutrient,
+    text: String,
+    updateText: (String) -> Unit
+): String {
+  var newText = text
+  try {
+    newText = removeLeadingZerosAndNonDigits(newText)
+    nutritionFields[index] = nutrient.copy(amount = newText.toDouble())
+    newText = newText.toDouble().toString()
+  } catch (_: NumberFormatException) {}
```

Comment:
You can Log it with a warning. Log.w("...", "_")

Reply:
will do ! Could be useful for debugging purposes